### ISSUE TITLE
Static nat hampers prive gateway access

### ISSUE THREAD

**Issue by DaanHoogland** [apache/cloudstack#6144]
Having two VPCs connected through their respective private gateways should allow instances from the one to connect to the other (given the proper firewall rules). When a static nat is assigned an instance in one of the VPC will stop that instance from being able to reach the other VPC.

##### ISSUE TYPE

 * Bug Report

##### COMPONENT NAME

~~~
networking
~~~

##### CLOUDSTACK VERSION

~~~
4.15.2
~~~

##### CONFIGURATION

VPC A:
    10.10.10.0/24
        PG-IP: 10.10.30.1
        PG-GW: 10.10.30.2
            tier 1: 10.10.10.0/25
                instance 1.a: 10.10.10.11

VPC B:
    10.10.20.0/24
        PG-IP: 10.10.30.2
        PG-GW: 10.10.30.1
            tier 1: 10.10.20.0/25
                instance 1.b: 10.10.20.22

##### STEPS TO REPRODUCE

~~~
create two VPCs
create a tier in each of them
deploy an instance in each of the tiers
create a private gateway in each of the VPCs with the same vlan and each others adresses as gateway
assign static routes for the other VPC in each of the PGs
ping the one instance fro the other (should work)
assign a static IP address (StaticNat) to the instance from which you ping 
restart the ping
~~~

##### EXPECTED RESULTS

~~~
ping works
~~~

##### ACTUAL RESULTS

~~~
ping go through
~~~

**Comment by andrijapanicsb** [comment]
We could argue that this is to the design (all outgoing packets going outside via StaticNat rules), but nevertheless, the more specific routes (Private Gateway routes) should take precedence.

**Comment by nvazquez** [comment]
Fixed by #6153
